### DISCUSSION TITLE
Passing additional params/data to routes?

### DISCUSSION THREAD

**Discussion by stefanisarie** [nuxt/nuxt#17460] [General]
Hey all!

I'm new to Nuxt3 thus the confusion for me... My case is as follow: I have a list of items:
```json
{
  "data": [
    {
      "id": 1,
      "internal_id": 5487,
      "name": "some name 1"
    },
    {
      "id": 2,
      "internal_id": 7158,
      "name": "some name 2"
    },
    {
      "id": 1,
      "internal_id": 58963,
      "name": "some name 3"
    }
  ]
}
```

And inside /pages folder the following route:
```bash
- pages
--- [internal_id]
------ [name].vue
```

In index.vue I'm building links to each entry. In "[name].vue" I want to query data by "id" instead of "internal_id" key. How do I pass additional/arbitrary params or data to routes? What I've tried so far and it didn't work (meta has no value in [name].js):

```js
ol
  li(v-for="entry in data")
    NuxtLink(:to="{ path: `/${entry.internal_id}/${entry.name}`, meta: { id: `${entry.id}`} }") {{entry.name}}
```

Basically, I want my routes to look like: https://somehost.com/:internal_id/:name but when some user is calling that route, I want to query my data by "id".

Hope it makes sense :)

Thank you!

**Comment by danielroe** [comment]
You can add items to the route meta with `definePageMeta`.

**Reply by stefanisarie** [reply]
Hey @danielroe !

Thank you for your reply! I've done it with cookies because https://v3.nuxtjs.org/api/utils/define-page-meta#definepagemeta is not yet ready. I've attached a click event on the link and set up a cookie with the id.

**Reply by danielroe** [reply]
Until that page is complete, you can see some docs here: https://v3.nuxtjs.org/guide/directory-structure/pages#page-metadata.
